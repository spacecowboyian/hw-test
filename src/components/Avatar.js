import React from 'react';
import { css } from 'emotion/macro';

import { colorCyan, spacing } from '../Styles';

const avatarCircle = css`
  display:flex;
  align-items: center;
  justify-content: center;
  background-color: ${colorCyan};
  width: 44px;
  height: 44px;
  border-radius: 44px;
  text-align: center;
  margin-right: calc(${spacing}*4);
`;

const avatarInitials = css`
  color: #006667;
  font-size: 18px;
  margin-top:-3px;
  margin-left:-2px;
`;

export default function Avatar(initials) {
  return (
    <div className={avatarCircle}><div className={avatarInitials}>{initials}</div></div>
  )
}
