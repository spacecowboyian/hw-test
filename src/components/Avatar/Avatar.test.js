import React from 'react';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Avatar from './Avatar';

afterEach(cleanup);

it('renders correctly', () => {
  const tree = renderer
    .create(<Avatar initials="IJ" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
