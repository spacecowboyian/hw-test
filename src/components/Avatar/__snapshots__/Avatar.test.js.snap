// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-tkappr-avatarCircle"
>
  <div
    className="css-bl0pin-avatarInitials"
  >
    IJ
  </div>
</div>
`;
